lib/using_restricted_array.rb

```diff
@@ -6,58 +6,109 @@
 
 # Calculates the length of the restricted array. All values are integers.
 # The restricted_array is terminated by 'nil' i.e. array[length] = nil
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n)
+# Space complexity: O(1)
 def length(array)
```

Comment:
👍

Reply:
> Great work, you hit all the learning goals and all the methods worked perfectly. You also had all the Big-O correct as well. Well done.

Thank you, Chris! I hope we'll have more practice in BigO.